RN2xxx.device.nut

```diff
@@ -63,94 +67,128 @@ class RN2xxx {
         _reset = reset;
         _uart = uart;
         _clearBuffer();
+        _sendQueue = [];
 
         _reset.configure(DIGITAL_OUT, 1);
     }
 
     function init(banner, cb = null) {
-        // Set init flag
-        _init = true;
-        // Set init callback
-        _initCB = cb;
+        // Set reset flag
+        _inReset = true;
+        // Set callback
+        _resetCB = cb;
         // Set banner
         _banner = banner;
 
-        // Reset device
+        // Hold reset pin low (active)
         _reset.write(0);
-        // Start initialization timeout timer
-        _timeout = imp.wakeup(INIT_TIMEOUT, _initTimeoutHandler.bindenv(this));
-        // Configure UART
+        // Start timeout timer
+        _timeout = imp.wakeup(RESET_TIMEOUT, _initTimeoutHandler.bindenv(this));
+        // Configure UART, _uartReceive handler will receive a banner to check
         _uart.configure(BAUD_RATE, WORD_SIZE, PARITY_NONE, STOP_BITS, NO_CTSRTS, _uartReceive.bindenv(this));
-        // Release Reset pin
+        // Release reset pin
         _reset.write(1);
     }
 
-    function hwReset() {
+    function hwReset(cb = null) {
+        // Toggle reset flag
+        _inReset = true;
+        _resetCB = cb;
         _reset.write(0);
+        // Start timeout timer
+        _timeout = imp.wakeup(RESET_TIMEOUT, _initTimeoutHandler.bindenv(this));
```

Comment:
should we `_cancelTimer(_timeout)` before? (just curious, not sure if this can be called multiple times)